artifacts/definitions/Windows/Forensics/BulkExtractor.yaml

```diff
@@ -0,0 +1,163 @@
+    or C:\\Folder\\Path"  
+    TargetAllPhysical - boolean option to include all attached physical disks  
+    TargetVSS - boolean option to target all VSC  
+    CarveEvtx - boolean option to include evtx carving  
+    FindRegex - regex to include for BulkExtractor find plugins  
+    
+    FreeCommand - supersedes evtx or find options and allows free form switch 
+    generation for adlib use cases. 
+    e.g '-E evtx, -e zip -S unzip_carve_mode=2'"  
+    Will add: 
+    command prefix: "-q 99999999999 -R'" and 
+    postfix: "-o [Outfolder] [Target]".  
+    To make: bulk_extractor q 99999999999 -R -E evtx, -e zip -S unzip_carve_mode=2 -o [outfolder] [Target]
+    
+    If FindRegex or "-f" has been used in FreeCommand the artifact will attempt 
+    to parse find.txt output.  
+
+    **Note**  
+    1. This artifact uses the 3rd party binaries feature. You will need to run 
+    Server.Utils.DownloadBinaries on the server prior to execution.  
+    2. Currently only supported for x64 bit machines.
+    3. This artifact usually takes a long time. Ensure default timeout is high 
+    enough for completion.  
+    4. This content is NOT recommended for hunting without great consideration as 
+    bulk_extractor is a multithreaded tool and utilises all CPU available on the 
+    endpoint.
+    
+reference:
+  - http://www.kazamiya.net/en/bulk_extractor-rec
+  - http://downloads.digitalcorpora.org/downloads/bulk_extractor/BEUsersManual.pdf
+  - https://simson.net/clips/academic/2013.COSE.bulk_extractor.pdf
+    
+author: Matt Green - @mgreen27
+ 
+precondition: SELECT OS From info() where OS = 'windows'
+
+parameters:
+  - name: binaryURL
+    description: |
+      Specify this as the base of the binary store (if empty we use
+      the server's public directory).
+  - name: Target
+    description: "Target. Can by physical drive, \\\\?\\HarddiskVolumeShadowCopy1 or C:\\Folder\\Path"
+    default: \\.\PhysicalDrive0
+  - name: TargetAllPhysical
+    description: "Target all attached physical drives"
+    type: bool
+  - name: TargetVSS 
+    description: "Target all VSC. Note: Not targeted to folder. Velociraptor CAN collect from the Volume Shadow direct targeted to folder with ntfs accessor so there may be a better way."
+    type: bool
+  - name: CarveEvtx
+    description: "Carve EVTX files"
+    type: bool
+  - name: FindRegex
+    description: "Regex for Bulk_extractor find plugin"
+  - name: FreeCommand
+    description: "Bulk_extractor custom commands. .e.g '-E evtx, -e zip -S unzip_carve_mode=2'"
+   
+sources:
+  - queries:
+    - LET bin <= SELECT * 
+        FROM Artifact.Generic.Utils.FetchBinary(
+            binaryURL=binaryURL, ToolName="Bulk_Extractor")
+    - LET tempfolder <= tempdir
+    - LET zip_file <= SELECT 
+            copy(filename=FullPath, dest=tempfile(extension=".exe"), 
+                accessor='zip') AS Payload 
+        FROM glob(globs="file:///" + (bin[0]).FullPath + 
```

Comment:
use url() to build this up? `url(path=bin[0].FullPath, fragment="bulk_extractor.exe").String`

Also if you are not having a glob (i.e. the filename is already known and cantt change (including casing) then you don't even need that - just copy it right off:
```
LET zip_file <= copy(filename=url(path=bin[0].FullPath, 
      fragment="bulk_extractor.exe").String, accessor='zip',
      dest=....)
```

Reply:
done